src/lib/swapper/swappers/ThorchainSwapper/endpoints.ts

```diff
@@ -146,9 +154,19 @@ export const thorchainApi: SwapperApi = {
         const publicClient = viemClientByChainId[chainId as EvmChainId]
         assert(publicClient !== undefined, `no public client found for chainId '${chainId}'`)
 
+        const expectedAmountOut =
+          BigInt(longtailData?.longtailToLExpectedAmountOut ?? 0) || BigInt(0)
+        // Ensure we have this to prevent sandwich attacks on the first step of a LongtailToL1 trade.
+        assert(expectedAmountOut !== undefined, 'expectedAmountOut is undefined')
```

Comment:
Check for `isZero()` as well?

Though, we may also want to move the comment above `amountOutMin` and do the check on it vs. `expectedAmountOut`. The reason for that is, quotes *may* look sane and the expectedAmountOut as far as the quote is concerned may be legit, but the sandwich/MEV happens at execution time on the EVM, so this being non-zero isn't what we care about.

That's *effectively* correct since as part of this flow, *if* it is zero, then `amountOutMin` would also be zero, but I'd still err on the side of technical correctness here and do the checks/comment on `amountOutMin`, since this is the parameter used as a directive to prevent this

Reply:
Agree! Changed the assertion to check `amountOutMin` as a part of https://github.com/shapeshift/web/pull/6052/commits/436e8b7537e640938e5900bcec78221090702f70.

Also added some extra paranoia checks in https://github.com/shapeshift/web/pull/6052/commits/353bad60b98761bfed788a99c7776979e74f641c.